rpc/wrtc_client_channel.go

```diff
@@ -223,7 +225,16 @@ func (ch *webrtcClientChannel) newClientStream(ctx context.Context, method strin
 }
 
 func makeRequestHeaders(ctx context.Context, method string) *webrtcpb.RequestHeaders {
-	headersMD, _ := metadata.FromOutgoingContext(ctx)
+	headersMD, ok := metadata.FromOutgoingContext(ctx)
+	if !ok {
+		headersMD = metadata.New(map[string]string{})
+	}
+
+	span := trace.FromContext(ctx)
```

Comment:
I think we may want to support this as a client and server interceptor so that we can use this not just with webrtc but grpc as well. https://gist.github.com/charithe/b5956620fa6bb0caf668f857d70a5fc4 may be useful as well

Reply:
Okay, I think the change I made addresses that. I tested the change by adding `rpc.WithUnaryClientInterceptor(rpc.UnaryClientTracingInterceptor())` to the DialOptions on the client side and the equivalent for the server side and it behaved as expected: [GCP Trace](https://console.cloud.google.com/traces/list?project=av-canary&tid=a22b0033d5e85c64f3e4ac10e79d59e5&pageState=(%22traceIntervalPicker%22:(%22groupValue%22:%22PT6H%22,%22customValue%22:null))&start=1687954823333&end=1687976423333)

Reply:
Sure, I'll look into that

Reply:
I've got a few ideas of how I could approach unit testing for this fix but I'm not sure exactly what you were looking for. I could:
1. Test the interceptors in isolation. That is, pass a handler to the interceptor which checks that the interceptor functioned as intended (created a new Span or injected the tracing metadata into the context)
2. Test the interceptors as part of a webrtc/grpc communication. That would mean constructing a whole client and server to facilitate sending and receiving messages and inspecting program behavior along the way, like what you did for [this server test](https://github.com/viamrobotics/goutils/blob/main/rpc/server_test.go), but stripping away authentication and other details. That feels more like an integration test than a unit test, and I don't know if that much boilerplate is warranted for verifying the interceptors' behavior when (1) is also an option, but I will if you think it's appropriate.
3. Another option?
I looked through the tests in the same directory I'm working in and they don't touch on inspecting interceptor behavior because that functionality is usually provided through the grpc library itself. I'm not sure if we ever define our own interceptors like this and test their functionality like I'm trying to, but if you know of a test you can point me to, I can model my tests after that

Reply:
I think option 2 is a great choice! RDK session and opid tests have good examples of testing these with unary and steam interceptors. Message me on slack if you wanna work through more details!